### PR TITLE
zotaut: Track erroneous records in DB

### PR DESCRIPTION
**Pull request by mtrojan-ub**

### CODE REVIEW

File: cpp/lib/src/ZoteroHarvesterUtil.cc [hunk 1]
```diff
@@ -499,50 +546,85 @@ time_t UploadTracker::getLastUploadTime(const unsigned zeder_id, const Zeder::Fl
...
-    if (recordAlreadyDelivered(hash, urls, &db_connection))
-        return false;
+    // Try to find an existing record & update it if possible
+    std::vector<Entry> entries;
+    if (recordAlreadyDelivered(hash, urls, /*delivery_states_to_ignore = */ {}, &entries, &db_connection)) {
+        const Entry * error_entry(nullptr);
+        for (const auto &entry : entries) {
+            if (entry.delivery_state_ != ERROR)
+                return false;
+            error_entry = &entry;
```
Comment (ruschein) [inline cpp/lib/src/ZoteroHarvesterUtil.cc:592]: I am not sure of this works. (We need to talk about this!)

**Review by ruschein — CHANGES_REQUESTED** [review]

File: cpp/data/installer/ub_tools.sql [hunk 2]
```diff
@@ -52,11 +51,13 @@ CREATE TABLE delivered_marc_records (
     id INT AUTO_INCREMENT PRIMARY KEY,
     hash VARCHAR(40) NOT NULL,
     zeder_journal_id INT(11) UNSIGNED NOT NULL,
+    delivery_state ENUM('automatic', 'manual', 'error') DEFAULT 'automatic',
```
[the remark this answers is not in the document]
Comment (mtrojan-ub) [reply cpp/data/installer/ub_tools.sql:54]: right, should be NOT NULL

File: cpp/cgi_progs/zotero_cgi.cc [hunk 3]
```diff
@@ -227,20 +230,16 @@ void ParseConfigFile(const std::multimap<std::string, std::string> &cgi_args, Te
             const auto zeder_journal_id(GetZederJournalId(zeder_id, zeder_instance, db_connection));
             std::string harvest_status("NONE");
```
[the remark this answers is not in the document]
Comment (mtrojan-ub) [reply cpp/cgi_progs/zotero_cgi.cc]: done

File: cpp/cgi_progs/zotero_cgi.cc [hunk 4]
```diff
@@ -494,29 +493,36 @@ void ProcessShowDownloadedAction(const std::multimap<std::string, std::string> &
     const std::string zeder_id(GetCGIParameterOrDefault(cgi_args, "zeder_id"));
     const std::string group(GetCGIParameterOrDefault(cgi_args, "group"));
     const Zeder::Flavour zeder_flavour(group == "IxTheo" or group == "RelBib" ? Zeder::Flavour::IXTHEO : Zeder::Flavour::KRIMDOK);
-    const std::string zeder_instance(group == "IxTheo" or group == "RelBib" ? "ixtheo" : "krimdok");
+    const std::string zeder_instance(GetZederInstanceForGroup(group));
 
     Template::Map names_to_values_map;
+
+    std::string style_css;
+    FileUtil::ReadString(TEMPLATE_DIRECTORY + "style.css", &style_css);
```
[the remark this answers is not in the document]
Comment (mtrojan-ub) [reply cpp/cgi_progs/zotero_cgi.cc:501]: Then the path will have two slasles

[the remark this answers is not in the document]
Comment (ruschein) [reply cpp/cgi_progs/zotero_cgi.cc]: N slashes for N > 1 is equivalent to a single slash.

**Pull request merged**